### ISSUE TITLE
Facebook live streams won't load or play.

### ISSUE THREAD

**Issue by stephenwithav** [videojs/video.js#3481]
## Description

Facebook live streams won't work as sources.

http://jsbin.com/hizerulasa/edit?html,output
## Steps to reproduce
1. Find a live stream on https://facebook.com/livemap
2. Right-click on the video, click Show Video URL, copy the link.
3. Use a tool like `youtube-dl` to get the rtmps link. (e.g., `youtube-dl -g [fbvideourl]`)
4. Change source's `src` attribute to the url from step 3.
5. Click play.
### Expected

A video showing the facebook stream.
### Actual

Nothing is shown.
#### videojs

5.0
#### browsers

Chrome stable
#### OSes

Linux
### plugins

No.

**Comment by gkatsev** [comment]
Unfortuantely, I've been unable to find the rtmp stream urls for facebook live but I just got one of their mpd urls and it works just fine using videojs-contrib-dash.

**Comment by stephenwithav** [comment]
Here's a Brazilian radio stream:

rtmps://edgetee-lax3-1.xx.fbcdn.net:443/live-md/1143890009009599?api=1&o=frc1&a=AfC-7Hm2Abut6xkS

**Comment by mister-ben** [comment]
I think you'd need to rejig that as `rtmps://edgetee-lax3-1.xx.fbcdn.net:443/live-md?api=1&o=frc1&a=AfC-7Hm2Abut6xkS&1143890009009599 ` to account for how we parse URLs into connection and stream, but it's still not playing. Do Facebook use anything like SWF verification?

**Comment by nickygerritsen** [comment]
`rtmpdump` is also having issues with that URL (the original one from @stephenwithav). As in this is the complete output:

```bash
$ rtmpdump -r 'rtmps://edgetee-lax3-1.xx.fbcdn.net:443/live-md/1143890009009599?api=1&o=frc1&a=AfC-7Hm2Abut6xkS'
RTMPDump v2.4
(c) 2010 Andrej Stepanchuk, Howard Chu, The Flvstreamer Team; license: GPL
WARNING: You haven't specified an output file (-o filename), using stdout
Connecting ...
INFO: Connected...
Starting download at: 0.457 kB
0.457 kB / 0.00 sec
Download complete
```

**Comment by stephenwithav** [comment]
That url is no longer valid @nickygerritsen , which is why you're getting the error.

Try getting any live stream from:

https://www.facebook.com/livemap

e.g.,

https://www.facebook.com/TheMahiraKhanOfficial/videos/1760946107515783/ currently returns:

rtmps://edgetee-lax3-1.xx.fbcdn.net:443/live-md/1760946110849116?o=atn2&a=AfDdgEjAhwxsDCWE

**Comment by stephenwithav** [comment]
.mp4s

**Comment by stephenwithav** [comment]
Oops--clicked the wrong button.

**Comment by stephenwithav** [comment]
.mp4s are for non-live streams, which isn't what I'm hoping for.

**Comment by albertogasparin** [comment]
I'm also having this issue with v5.11. Looks like Fb `rtmps` is not played (just black screen), while [here](https://www.wowza.com/testplayers) they work as expected.
This url will be valid for the next 3 hours: 
```
rtmps://edgetee.xx.fbcdn.net:443/live-md/1282731655104706?api=1&s_p=1&o=lla2&a=AfDrx0WuwBQbFnDh
```
If you guys can find some time to debug what is going wrong that would be great (just keep in mind that Fb limits the number of connected clients to 2). When this expires I can eventually generate another one for you.

**Comment by stephenwithav** [comment]
Will do this weekend.  Thanks!

**Comment by gkatsev** [comment]
@albertogasparin thanks for PR. We'll have to consider whether we want to add this feature (probably, assuming it's safe), especially taking #3520 into account.

**Comment by albertogasparin** [comment]
Well, I agree with #3520, making Flash optional is a nice move as it becomes less and less supported. However, as the fix for this issue is Flash-side only so I don't see any reason for it being blocked by #3520. Moreover, rtmp will likely remain Flash only even in the future, so providing proper support for it would be a great achievement for VideoJS, while moving other streaming techniques to be HTML/JS only.

**Comment by Invictaz** [comment]
@stephenwithav I only get a .mpd manifest with youtube-dl. Seems the videos on Facebook are MPEG-DASH and not HLS M3U8. How do you get the RTMP? With Streamlink I can get the video but only 360P.
Seems that for higher resolutions you need to login?

**Comment by seisdr** [comment]
Hello

**Comment by seisdr** [comment]
Fixed or yet

**Comment by clinttowne1** [comment]
![2541EE54-0216-4083-8746-9AEBF3784369_1_105_C](https://user-images.githubusercontent.com/60196739/73042883-ecc66a00-3e30-11ea-8d32-0d877986bf2a.JPG)

**Comment by gkatsev** [comment]
Given that Adobe Flash is [being end-of-lifes this month](https://www.adobe.com/products/flashplayer/end-of-life.html) we will no longer be working on flash related features.
If they have an HLS or DASH output, we could theoretically play it but we will not be investigating RTMP anymore.

**Comment by sebseb7** [comment]
this was about facebook live, not about flash or rtmp.

**Comment by gkatsev** [comment]
Unless I'm mistaken, this was about playing back rtmp that came back from facebook live.

**Comment by sebseb7** [comment]
facebook live streams are dash not rtmp

**Comment by gkatsev** [comment]
All the comments above are talking about playing back RTMP that is coming from Facebook Live and refer to a pullrequest against the video-js-swf repo (https://github.com/videojs/video-js-swf/pull/208). If Facebook Live produces DASH streams now that's great! Video.js can play back DASH natively.
If there are Facebook Live related issues when playing back their DASH streams, they should be opened up as a new issue, please.

**Comment by sebseb7** [comment]
the bug was opened because their DASH streams do not play with videojs, but honestly I don't care. I use dash.js

**Comment by gkatsev** [comment]
The original test case provided includes only rtmp urls (http://jsbin.com/hizerulasa/edit?html,output) so I'm not following where DASH comes into play for *this* issue.

**Comment by sebseb7** [comment]
"youtube-dl -g [fbvideourl]" returna a dash URL these days.

**Comment by gkatsev** [comment]
Good to know. That must've changed since this issue was opened. I hope that RTMP isn't delivered to browsers nowadays at all.
